### PR TITLE
Automatic addition of the OTel Logback appender by the OTel starter

### PR DESCRIPTION
**Pull request by jeanbisutti**
With this PR the OTel starter automatically adds the OTel Logback appender if the appender is not configured in an XML file

### CODE REVIEW

File: smoke-tests-otel-starter/src/main/resources/logback.xml [hunk 1]
```diff
@@ -1,12 +0,0 @@
...
-
-  <appender name="OpenTelemetry"
-            class="io.opentelemetry.instrumentation.logback.appender.v1_0.OpenTelemetryAppender">
-  </appender>
-
-  <root level="INFO">
-    <appender-ref ref="OpenTelemetry"/>
-  </root>
-
-</configuration>
```
Comment (trask) [inline smoke-tests-otel-starter/src/main/resources/logback.xml:12]: can we test both with and without xml configuration? if not possible, I think it's ok as long as we test one way here and the other way in the smoke test

Comment (jeanbisutti) [reply smoke-tests-otel-starter/src/main/resources/logback.xml:12]: A test is executed with an XML file [here](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/e448478f9f419d6870ac3853e2448be8658dc41b/instrumentation/spring/spring-boot-autoconfigure/src/testLogbackAppender/java/io/opentelemetry/instrumentation/spring/autoconfigure/instrumentation/logging/LogbackAppenderTest.java#L49). [It checks that the OpenTelemetry appender is added only once.](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/e448478f9f419d6870ac3853e2448be8658dc41b/instrumentation/spring/spring-boot-autoconfigure/src/testLogbackAppender/java/io/opentelemetry/instrumentation/spring/autoconfigure/instrumentation/logging/LogbackAppenderTest.java#L63).

The Logback XML file was removed from the OpenTelemetry starter smoke test.

You mean to test both configurations in the smoke tests?

Comment (trask) [reply smoke-tests-otel-starter/src/main/resources/logback.xml:12]: oh, sorry, I missed (again) that this was under smoke-test 😅

**Review by trask — APPROVED** [review]

**Comment by denodaeus** [comment]
How can I prevent this from "automagically" breaking my app if I choose to not use logback but instead something like log4j2?

I've followed the instructions for adding log4j2 as the log appender, but I continuously get this "magic":

22:05:06.622 [main] ERROR com.my.app.MyApplication - PortfolioManagementServiceApplication failed to start
java.lang.IllegalArgumentException: Cannot instantiate interface org.springframework.context.ApplicationListener : io.opentelemetry.instrumentation.spring.autoconfigure.instrumentation.logging.LogbackAppenderApplicationListener
	at org.springframework.boot.SpringApplication.createSpringFactoriesInstances(SpringApplication.java:449) ~[spring-boot-2.7.8.jar:2.7.8]

**Comment by trask** [comment]
> java.lang.IllegalArgumentException: Cannot instantiate interface org.springframework.context.ApplicationListener : io.opentelemetry.instrumentation.spring.autoconfigure.instrumentation.logging.LogbackAppenderApplicationListener

hi @denodaeus! can you open a new issue for this so that it's not lost? thanks

**Pull request merged**